corehq/apps/locations/adjacencylist.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import absolute_import
+from django.db.models import IntegerField
+from django.db.models.expressions import Value
+from django.db.models.query import Q, QuerySet
+from django_cte import With
+from mptt.models import MPTTModel, TreeManager
+
+from .queryutil import ComparedQuerySet, TimingContext
+
+int_field = IntegerField()
+
+
+class AdjListManager(TreeManager):
+
+    def cte_get_ancestors(self, node, ascending=False, include_self=False):
+        """Query node ancestors
+
+        :param node: A model instance or a QuerySet or Q object querying
+        the adjacency list model. If a QuerySet, it should query a
+        single value with something like `.values('pk')`. If Q the
```

Comment:
if it's a single value, does `flat` need to be explicitly `True` or `False`, or it doesn't matter?

Reply:
I don't think `flat` is a valid parameter to `.values(...)`. It's definitely not required, as the query from the given queryset will be embedded into the `WHERE` clause of the new queryset returned by this method.

Reply:
right nevermind, i had read this as values_list